### PR TITLE
#289 - Laravel 10 update

### LINKED ISSUE #289: Upgrade to Laravel 10
Reminder to upgrade project to Laravel 10. But it's some blockers here, like rackbeat/laravel-ui-avatars package that is not supporting Laravel 10 yet.

### LINKED ISSUE #293: Tests should not be dependent on day of week
**Bug description**
> Two of three tests from `SendVacationRequestSummariesToApprovers` are failing because command `SendVacationRequestSummariesToApprovers` is ran there and its `shouldHandle()` method relies on weekdays.

**Reproduction steps**
> run `dcr -e XDEBUG_MODE=off php php artisan test --filter SendVacationRequestSummariesTest` on Saturday

**Expected behavior**
> Tests should have the same outcome for every day.

### PR DESCRIPTION
**Pull request by krzysztofrewak**
In this PR I updates Laravel to 10.4, Codestyle to 2.0 and PHP to 8.2.

This should close #289 and should close #293.
This will probably automatically close #287 and #288.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -11,29 +11,28 @@
         "azuyalabs/yasumi": "^2.4",
         "barryvdh/laravel-dompdf": "^2.0",
         "doctrine/dbal": "^3.3",
-        "fruitcake/laravel-cors": "^3.0",
         "guzzlehttp/guzzle": "^7.0.1",
         "inertiajs/inertia-laravel": "^0.6.2",
-        "laravel/framework": "^9.7",
-        "laravel/sanctum": "^3.0",
+        "laravel/framework": "^10.0",
```
Comment (mtracz) [inline composer.json:16]:
```suggestion
        "laravel/framework": "^10.4.0",
```

File: app/Infrastructure/Console/Kernel.php [hunk 2]
```diff
@@ -27,14 +38,15 @@ protected function schedule(Schedule $schedule): void
         $schedule->job(CheckYearPeriod::class)
             ->yearlyOn(1, 1, "01:00");
 
-        $this->scheduleDatabaseBackup($schedule);
-    }
+        $schedule->command("cache:prune-stale-tags")->hourly();
```
Comment (mtracz) [inline app/Infrastructure/Console/Kernel.php:41]: I would like to use class name instead of raw string command signature.

**Review by mtracz — CHANGES_REQUESTED** [review]

**Comment by mtracz** [comment]
Please remove `webpack.config.js`, `webpack.mix.js` from production Dockerifle.

**Comment by mtracz** [comment]
Also validation language translations are outdated.

https://github.com/laravel/framework/blob/10.x/src/Illuminate/Translation/lang/en/validation.php

File: app/Infrastructure/Console/Kernel.php (hunk 2 above)
Comment (krzysztofrewak) [reply app/Infrastructure/Console/Kernel.php]: Changed.

**Review by mtracz — APPROVED** [review]

**Comment by Baakoma** [comment]
Please, change the xDebug version to 3.2.0, because current version doesn't support php 8.2:

> pecl/xdebug requires PHP (version >= 7.2.0, version <= 8.1.99), installed version is 8.2.4

File: package.json [hunk 3]
```diff
@@ -22,9 +22,7 @@
         "echarts": "^5.4.1",
         "flatpickr": "^4.6.13",
         "laravel-vite-plugin": "^0.7.3",
-        "lodash": "^4.17.21",
         "luxon": "^3.2.1",
-        "postcss": "^8.4.21",
```
Comment (Baakoma) [inline package.json:27]: Why these packages are removed from package.json? I'm pretty sure we use them in the project.

Comment (krzysztofrewak) [reply package.json]: I removed it accordingly to the upgrade guide recommendations: https://github.com/laravel/laravel/compare/9.x...10.x They are still in node modules as they're required by other packages.

Anyway, I'm returning them back as I cna see that lodash is used directly in some files.

**Review by Baakoma — APPROVED** [review]

**Review by mtracz — APPROVED** [review]

**Pull request merged**